packages/jest-cli/src/watch.js

```diff
@@ -33,46 +34,48 @@ const watch = (
   stdin?: stream$Readable | tty$ReadStream = process.stdin
 ) => {
   setWatchMode(argv, argv.watch ? 'watch' : 'watchAll', {
-    pattern: argv._,
+    testNamePattern: argv.testNamePattern,
+    testPathPattern: argv.testPathPattern
+      || argv._ instanceof Array
+        ? argv._.join('|')
+        : '',
```

Comment:
nit, code style:

```
testPathPattern: argv.testPathPattern || Array.isArray(argv._)
  ? argv._.join('|')
  : '',
```

do you also want to  check for `.length` of the array? I guess it produces the same output here.

Don't use instanceof Array, `Array.isArray` is much safer.